### PR TITLE
[cxx-interop] Use the locations imported from C++

### PR DESCRIPTION
**Pull request by Xazax-hun**
A recent PR (#77204) started to import C++ source locations into Swift. This PR flips a switch so these locations are actually used more widely.

### CODE REVIEW

File: test/Interop/Cxx/class/nonescapable-errors.swift [hunk 1]
```diff
@@ -48,6 +48,7 @@ public func noAnnotations() -> View {
     // CHECK-NOT: nonescapable.h:19
     f2(nil, nil)
     // CHECK: nonescapable.h:23:6: warning: the returned type 'View' is annotated as non-escapable; its lifetime dependencies must be annotated
+    // CHECKL nonescapable.h:23:6: error: cannot infer lifetime dependence, no parameters found that are either ~Escapable or Escapable with a borrowing ownership
```
Comment (Xazax-hun) [inline test/Interop/Cxx/class/nonescapable-errors.swift:51]: Previously, this error had unknown location.

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Review by egorzhdan — APPROVED** [review]

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
Please test with following pull request:
https://github.com/swiftlang/llvm-project/pull/9538

@swift-ci please smoke test

**Pull request merged**